MonMaraicher/Features/Map/MapView.swift

```diff
@@ -14,23 +14,32 @@ struct MapView: View {
 
     var body: some View {
         Map(position: $viewModel.mapCameraPosition, selection: $viewModel.selectedMarker) {
-
             ForEach(viewModel.allMarkers, id: \.id) { marker in
                 Marker(marker.title, systemImage: marker.systemImage, coordinate: marker.coordinate)
                     .tag(marker)
                     .tint(.orange)
             }
-
             UserAnnotation()
-
         }
         .mapStyle(.standard(elevation: .realistic))
         .mapControls {
             MapUserLocationButton()
         }
+        .overlay(alignment: .center) {
+            if viewModel.farmersLoadingInProgress && viewModel.hasUserAcceptedLocation {
```

Comment:
I would add a computed property in your viewModel returning that value instead of doing it in your view